src/Symfony/Component/Messenger/Transport/Doctrine/DoctrineReceiver.php

```diff
@@ -46,6 +49,17 @@ public function get(): iterable
     {
         try {
             $doctrineEnvelope = $this->connection->get();
+            $this->retryingSafetyCounter = 0; // reset counter
+        } catch (RetryableException $exception) {
+            // Do nothing when RetryableException occurs less than "MAX_RETRIES"
+            // as it will likely be resolved on the next call to get()
+            // Problem with concurrent consumers and database deadlocks
+            if (++$this->retryingSafetyCounter >= self::MAX_RETRIES) {
+                $this->retryingSafetyCounter = 0; // reset counter
+                throw new TransportException($exception->getMessage(), 0, $exception);
+            }
+
+            return [];
```

Comment:
Wouldn't it be better to retry "internally" (i.e. `return $this->get()`) instead of returning this empty array? Otherwise, in the event of using multiple transports, the worker might start consuming lower priority messages, which would make no sense. Similar with _whatever_ might decorate this receiver... it would "see" an empty array while actually, it was an error.

Reply:
Does not seem better to me. This way we can use the delay that is added after the get which is what we want with a deadlock/timeout problem.